src/Payum/Core/Registry/DynamicRegistry.php

```diff
@@ -64,7 +77,10 @@ public function getGateway($name)
      */
     public function getGateways()
     {
-        return $this->staticRegistry->getGateways();
+        return array_replace(
```

Comment:
I am going to completely remove static registry from here, if you need both of them you have to use https://github.com/Payum/Payum/blob/master/src/Payum/Core/Registry/FallbackRegistry.php

The builder already works this way.